### DISCUSSION TITLE
how can I load the data for a customPivotEntity when I load my entities ?

### DISCUSSION THREAD

**Discussion by BrannJoly** [mikro-orm/mikro-orm#4414] [Q&A] [answered]
In one of my entities I have this relationship :

```
@Entity({ tableName: 'projects' })
export class ProjectDbEntity {
...
  @ManyToMany({ entity: () => TargetDbEntity, pivotEntity: () => TargetProjectsDbEntity })
  targets = new Collection<TargetDbEntity>(this);
}

@Entity()
export class TargetProjectDbEntity {
  @ManyToOne({ primary: true, entity: 'ProjectDbEntity' })
  project: ProjectDbEntity;

  @ManyToOne({ primary: true })
  target: TargetDbEntity;

  @Property({ nullable: true, type: 'float' })
  value?: number;
}

```

when I load a ProjectDbEntity, I can access the targets collection just fine, but I don't know how to get the associated value. 
I know I can get it explicitely using em.find('TargetProjectsDbEntity', { project: 123, target: 321 }) but I'd like to have mikro orm preload everything when I retrieve the entity from the base, a bit like when you use 'populate' to initialize a collection

Is this usecase supported ? How ?
Thanks

**Comment by B4nan** [comment, marked as answer]
If you want to preload the values when populating the M:N, you will need to define the pivot relations (the 1:m) and use those, that's the only way to actually have the pivot entities created and available. When you work with the M:N directly, it only uses the pivot entity to infer the join columns, but still does a simple FK lookup through it, not creating any pivot entities behind the scenes.

https://mikro-orm.io/docs/next/composite-keys#use-case-3-join-table-with-metadata

**Reply by BrannJoly** [reply]
thank you
